src/main/java/org/truffleruby/core/symbol/SymbolTable.java

```diff
@@ -96,6 +97,10 @@ public DynamicObject getSymbol(String string) {
 
     @TruffleBoundary
     public DynamicObject getSymbol(Rope rope) {
+        if (rope instanceof NativeRope) {
+            rope = ((NativeRope) rope).toLeafRope();
+        }
```

Comment:
What's the issue to lookup a Symbol with a NativeRope?
The Rope used in the symbolMap is the one returned by RopeCache.getRope which is built from a managed `byte[]`.

Reply:
The immediate problem is `NativeRope#withEncoding` throws an exception, which affects the `if` statement following this one. I can fix that, but I think it's better to fix the issue here.

But I think we may have another issue where the `ropeKey` stored in the symbol uses the original rope, not the one built with a managed `byte[]`.